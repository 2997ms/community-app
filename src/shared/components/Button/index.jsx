/**
 * Button component which accepts via props the text to render
 * inside the button, its size and style.
 *
 * On hover, mouse cursor should become pointer, and the onClick()
 * callback passed from parent should be triggered on click.
 *
 * Style:
 * To create outline btn pass 'tc-outline-btn' className
 * To create filled btn pass '{color} tc-blue-btn' className. eg. 'red tc-blue-btn'
 * To create big btn pass 'tc-bg-btn' className along with one of the above two btn type
 */

import React from 'react';
import PT from 'prop-types';
import _ from 'lodash';
import './styles.scss';

export default function Button(props) {
  return (
    <button
      className={`tc-btn ${props.className}`}
      onClick={() => props.onClick()}
    >
      {props.children}
    </button>
  );
}

Button.defaultProps = {
  className: 'tc-blue-btn',
  onClick: _.noop,
  children: null,
};

Button.propTypes = {
  className: PT.string,
  onClick: PT.func,
  children: PT.node,
};
