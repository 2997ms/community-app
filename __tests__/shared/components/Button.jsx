import Button from 'components/Button';
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';

test('renders correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Button />);
  const button = renderer.getRenderOutput();
  expect(button).toMatchSnapshot();
});
