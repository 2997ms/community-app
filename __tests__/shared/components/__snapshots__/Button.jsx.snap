// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="tc-btn tc-blue-btn"
  onClick={[Function]}
/>
`;
